src/safeposix/syscalls/fs_calls.rs

```diff
@@ -389,42 +389,108 @@ impl Cage {
+    /// [mknod(2)](https://man7.org/linux/man-pages/man2/mknod.2.html)
+    ///
     pub fn mknod_syscall(&self, path: &str, mode: u32, dev: u64) -> i32 {
-        //Check that path is not empty
+        // Return an error if the provided path is empty
         if path.len() == 0 {
             return syscall_error(Errno::ENOENT, "mknod", "given path was null");
         }
+        // Retrieve the absolute path from the root directory. The absolute path is
+        // then used to validate directory paths while navigating through
+        // subdirectories and establishing new directory at the given location.
         let truepath = normpath(convpath(path), self);
 
-        //pass the metadata to this helper. If passed table is none, then create new instance
+        // Store the FileMetadata into a helper variable which is used for fetching
+        // the metadata of a given inode from the Inode Table. 
         let metadata = &FS_METADATA;
 
+        // Walk through the absolute path which returns a tuple consisting of inode
+        // number of file (if it exists), and inode number of parent (if it exists)
         match metawalkandparent(truepath.as_path()) {
-            //If neither the file nor parent exists
-            (None, None) => syscall_error(
-                Errno::ENOENT,
-                "mknod",
-                "a directory component in pathname does not exist or is a dangling symbolic link",
-            ),
-
-            //If the file doesn't exist but the parent does
+            // Case: When the file doesn't exist but the parent directory exists
             (None, Some(pardirinode)) => {
-                let filename = truepath.file_name().unwrap().to_str().unwrap().to_string(); //for now we assume this is sane, but maybe this should be checked later
-
-                //assert sane mode bits (asserting that the mode bits make sense)
+                // for now we assume this is sane, but maybe this should be checked later
+                let filename = truepath.file_name().unwrap().to_str().unwrap().to_string(); 
+
+                // S_FILETYPEFLAGS represents a bitmask that can be used to extract
+                // the file type information from a file's mode.
+                // This code is referenced from Lind-Repy codebase.
+                // Here, we are checking whether the mode bits are sane by ensuring
+                // that only valid file permission bits (S_IRWXA) and file type bits
+                // (S_FILETYPEFLAGS) are set. Else, we return the error.
                 if mode & (S_IRWXA | S_FILETYPEFLAGS as u32) != mode {
                     return syscall_error(Errno::EPERM, "mknod", "Mode bits were not sane");
                 }
+
+                // As of now, the only file type in LIND supported by mknod syscall
+                // is "Char Device".
+                // In order to check for Char file type, a bitwise-AND operation is
+                // performed with the "mode".
```

Comment:
is S_IFCHR part of the filetype bitmask? if so can be add a comment mentioning that

Reply:
Updated the comment section.